src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/lib/__tests__/get_matched_indices.test.js

```diff
@@ -51,6 +51,52 @@ describe('getMatchedIndices', () => {
     });
   });
 
+  it('should support queries with wildcards in various places', () => {
+    const indices = [
+      { name: 'kibana' },
+      { name: 'es' },
+      { name: '.kibana' }
+    ];
+
+    const query = 'k*ib*';
+    const matchedIndices = [
+      { name: 'kibana' },
+    ];
+
+    const result = getMatchedIndices(indices, matchedIndices, query, false);
+
+    expect(result).toEqual({
+      allIndices: [{ name: 'kibana' }, { name: 'es' }],
+      exactMatchedIndices: [{ name: 'kibana' }],
+      partialMatchedIndices: [{ name: 'kibana' }],
+      visibleIndices: [{ name: 'kibana' }]
+    });
+  });
+
+  it('should support queries with wildcards in various places again', () => {
+    const indices = [
+      { name: 'kibana' },
+      { name: 'kibana2' },
+      { name: 'es' },
+      { name: '.kibana' }
+    ];
+
+    const query = '*kib*';
+    const matchedIndices = [
+      { name: 'kibana' },
+      { name: 'kibana2' }
+    ];
+
+    const result = getMatchedIndices(indices, matchedIndices, query, false);
+
+    expect(result).toEqual({
+      allIndices: [{ name: 'kibana' }, { name: 'kibana2' }, { name: 'es' }],
+      exactMatchedIndices: [{ name: 'kibana' }, { name: 'kibana2' }],
+      partialMatchedIndices: [{ name: 'kibana' }, { name: 'kibana2' }],
+      visibleIndices: [{ name: 'kibana' }, { name: 'kibana2' }]
```

Comment:
Something to think about in a later iteration, but I think we might be able to make these tests clearer if we pared the tests down to make assertions against _just_ `allIndices`, `exactMatchedIndices`, `partialMatchedIndices`, and `visibleIndices`. This would result in more focused tests and I think it'd be easier to spot the relationships between the input and each specific type of output. It would result in more tests, however. What do you think?

Reply:
Totally agree. Great suggestion!